include/El/core/Matrix/impl_cpu.hpp

```diff
@@ -18,255 +18,231 @@ namespace El
 // Constructors and destructors
 // ============================
 
-template<typename Ring>
-Matrix<Ring, Device::CPU>::Matrix() { }
+template <typename T>
+Matrix<T, Device::CPU>::Matrix() { }
 
-template<typename Ring>
-Matrix<Ring, Device::CPU>::Matrix(Int height, Int width)
-    : AbstractMatrix<Ring>{height,width,height}
+template <typename T>
+Matrix<T, Device::CPU>::Matrix(
+    size_type height, size_type width, size_type leadingDimension)
+    : AbstractMatrix<T>{height, width, leadingDimension}
```

Comment:
This should be `{height, width, max(height, leadingDimension)}` or something similar. `leadingDimension` defaults to 0; a later thing in `AbstractMatrix` checks `max(leadingDimension, 1)`. Thus, when `leadingDimension` is omitted, we end up with a leading dimension of 1 instead of `height`.

This leads to 💥: Consider `El::Matrix<uint8_t>(32, 1)`. The memory allocated with be `ldim*width == 1*width == 1`. 💥